src/libslic3r/GCodeWriter.cpp

```diff
@@ -315,7 +323,17 @@ std::string GCodeWriter::write_acceleration(){
     } else if (FLAVOR_IS(gcfMarlinFirmware) || FLAVOR_IS(gcfRepRap)) {
         // M204: Set printing & travel acceleration
         gcode << "M204 P" << m_current_acceleration << " T" << (m_current_travel_acceleration > 0 ? m_current_travel_acceleration : m_current_acceleration);
-    } else { // gcfMarlinLegacy
+    } else if (FLAVOR_IS(gcfKlipper)){
+        if (this->config.deceleration_match_to_er_acceleration == true){
```

Comment:
maybe just `if (this->config.deceleration_factor.value > 0)`

Reply:
that would make the checkbox pointless then wouldn't it?
i guess the tooltip could be updated to say "set to 0 to disable", that would help clean up the GUI a fraction

Reply:
yes. There is enough settings.
If there is a need to have a checkbox to deactivate, then it can be added by the gui itself on top.